### PR TITLE
Contexts edge cases

### PR DESCRIPTION
**Pull request by pmuir**

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 9abdb6a02e7b926f646823041ff90a1be9e2bd4c
  * Your commit message does not contain a body

 * Commit: 91f24e9b62d0b50dd8740c749cf93e23243324a0
  * Your commit message does not contain a body

Guidelines are available at https://github.com/fabric8io/fabric8-ui

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 931a7f305a1fb15df40bc83a1f2434de54454aaa
  * Your commit message does not contain a body

 * Commit: f7c25cd50db41a3fb8c29bf17ef05ebfde9edd68
  * Your commit message does not contain a body

Guidelines are available at https://github.com/fabric8io/fabric8-ui

---
This message was auto-generated by https://gitcop.com

File: src/app/header/header.component.ts [hunk 1]
```diff
@@ -32,8 +32,21 @@ export class HeaderComponent implements OnInit {
...
+        return Observable.combineLatest(
+          headerComponent.contexts.current.map(val => val.user.id),
+          headerComponent.userService.loggedInUser.map(val => val.id),
+          (a, b) => (a !== b)
+        );
+      }
+    ],
+    [
+      '_resources', function (headerComponent) {
+        return Observable.combineLatest(
```
Comment (joshuawilson) [inline src/app/header/header.component.ts]: This function and the one above it appear to do the same thing. Do you plan to make them different.

Comment (pmuir) [inline src/app/header/header.component.ts]: Yeah, they could be made to reference an external function

File: src/app/shared/context-current-user-auth-guard.service.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
-import { Context, Contexts } from 'ngx-fabric8-wit';
+import { Observable } from 'rxjs/Observable';
+import { Context, Contexts, Notification, Notifications, NotificationType } from 'ngx-fabric8-wit';
```
Comment (joshuawilson) [inline src/app/shared/context-current-user-auth-guard.service.ts]: It looks like you aren't using 'context' nor 'DummyService' anymore.

File: src/app/shared/context.service.ts [hunk 3]
```diff
@@ -1,10 +1,8 @@
-import { MenusService } from './../header/menus.service';
-import { MenuedContextType } from './../header/menued-context-type';
 import { Injectable } from '@angular/core';
 import { Router } from '@angular/router';
 
 import { Broadcaster, User, UserService, Entity } from 'ngx-login-client';
-import { Space, Contexts, Context, ContextType, ContextTypes, SpaceService } from 'ngx-fabric8-wit';
+import { Space, Contexts, Context, ContextType, ContextTypes, SpaceService, Notifications, Notification, NotificationType } from 'ngx-fabric8-wit';
```
Comment (joshuawilson) [inline src/app/shared/context.service.ts]: ContextTypes is not used.

File: src/app/shared/context.service.ts [hunk 4]
```diff
@@ -13,10 +11,13 @@ import { Observable } from 'rxjs';
 
 import { DummyService } from './../shared/dummy.service';
 import { Navigation } from './../models/navigation';
+import { MenusService } from './../header/menus.service';
+import { MenuedContextType } from './../header/menued-context-type';
```
Comment (joshuawilson) [inline src/app/shared/context.service.ts]: this does not appear to be used

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 931a7f305a1fb15df40bc83a1f2434de54454aaa
  * Your commit message does not contain a body

 * Commit: 3e7f85ab603e83db18a3b06e8e636f64b751413c
  * Your commit message does not contain a body

Guidelines are available at https://github.com/fabric8io/fabric8-ui

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 485fdffc8cbb641abbd0e6ef7509c529e82545ec
  * Your commit message does not contain a body

 * Commit: ebd507e1c61852b8d360aab858ab97473287d1bd
  * Your commit message does not contain a body

Guidelines are available at https://github.com/fabric8io/fabric8-ui

---
This message was auto-generated by https://gitcop.com

**Comment by joshuawilson** [comment]
When I tried to create a space with 'foo' it gave me a notification error that said it was undefined. When I changed it to 'foo22' worked. I had used 'foo' in the past but the message does not indicate that is the issue.

"Failed to create the collaboration space "foo" because undefined"

`"{"errors":[{"code":"bad_parameter","detail":"Bad value for parameter 'Name': 'foo' (expected: 'unique')","status":"400","title":"Bad parameter error"}]}↵{"errors":[{"code":"unknown_error","detail":"pq: Could not complete operation in a failed transaction","status":"500","title":"Unknown error"}]}`

from the request body I can see that it failed on uniqueness. But I don't have it as a space to choose any more, so I thought I could use it.

**Comment by pmuir** [comment]
Yes, we have an open issue for that https://github.com/fabric8io/fabric8-ui/issues/176

**Pull request merged**